packages/lib/src/spatial-navigation/components/ScrollView.tsx

```diff
@@ -15,41 +17,109 @@ type Props = {
   offsetFromStart?: number;
   children: React.ReactNode;
   style?: ViewStyle;
+  topArrow?: ReactNode;
+  bottomArrow?: ReactElement;
+  topArrowContainerStyle?: ViewStyle;
+  bottomArrowContainerStyle?: ViewStyle;
+  /** Number of pixels scrolled every 10ms */
+  scrollSpeed?: number;
 };
 
 export const SpatialNavigationScrollView = ({
   horizontal = false,
   style,
   offsetFromStart = 0,
   children,
+  topArrow,
+  bottomArrow,
+  topArrowContainerStyle,
+  bottomArrowContainerStyle,
+  scrollSpeed = 10,
 }: Props) => {
   const { scrollToNodeIfNeeded: makeParentsScrollToNodeIfNeeded } =
     useSpatialNavigatorParentScroll();
   const scrollViewRef = useRef<ScrollView>(null);
+  const {
+    deviceType,
+    getScrollingIntervalId: getScrollingId,
+    setScrollingIntervalId: setScrollingId,
+  } = useDevice();
+  const scrollY = useRef<number>(0);
 
   const scrollToNode = useCallback(
     (newlyFocusedElementRef: RefObject<View>) => {
       try {
-        newlyFocusedElementRef?.current?.measureLayout(
-          scrollViewRef?.current?.getInnerViewNode(),
-          (left, top) =>
-            scrollToNewlyFocusedElement({
-              newlyFocusedElementDistanceToLeftRelativeToLayout: left,
-              newlyFocusedElementDistanceToTopRelativeToLayout: top,
-              horizontal,
-              offsetFromStart,
-              scrollViewRef,
-            }),
-          () => {},
-        );
+        if (deviceType === 'remoteKeys') {
+          newlyFocusedElementRef?.current?.measureLayout(
+            scrollViewRef?.current?.getInnerViewNode(),
+            (left, top) =>
+              scrollToNewlyFocusedElement({
+                newlyFocusedElementDistanceToLeftRelativeToLayout: left,
+                newlyFocusedElementDistanceToTopRelativeToLayout: top,
+                horizontal,
+                offsetFromStart,
+                scrollViewRef,
+              }),
+            () => {},
+          );
+        }
       } catch {
         // A crash can happen when calling measureLayout when a page unmounts. No impact on focus detected in regular use cases.
       }
       makeParentsScrollToNodeIfNeeded(newlyFocusedElementRef); // We need to propagate the scroll event for parents if we have nested ScrollViews/VirtualizedLists.
     },
-    [makeParentsScrollToNodeIfNeeded, horizontal, offsetFromStart],
+    [makeParentsScrollToNodeIfNeeded, horizontal, offsetFromStart, deviceType],
   );
 
+  const onMouseEnterTop = () => {
```

Comment:
this all needs `useCallback` wrapping (and when you do, you should check that the values do change at relevant times) 😉